feature/gnmi/otg_tests/telemetry_basic_check_test/telemetry_basic_check_test.go

```diff
@@ -669,61 +673,76 @@ func TestIntfCounterUpdate(t *testing.T) {
+		SetAddress("198.51.100.3").SetGateway("198.51.100.2").
+		SetPrefix(31)
+
+	flowName := "telemetry_test_flow"
+	flowipv4 := config.Flows().Add().SetName(flowName)
+	flowipv4.Metrics().SetEnable(true)
+	flowipv4.TxRx().Device().
+		SetTxNames([]string{intf1.Name() + ".IPv4"}).
+		SetRxNames([]string{intf2.Name() + ".IPv4"})
+	flowipv4.Size().SetFixed(100)
+	flowipv4.Rate().SetPercentage(1)
+	e1 := flowipv4.Packet().Add().Ethernet()
+	e1.Src().SetValue(eth1.Mac())
+	v4 := flowipv4.Packet().Add().Ipv4()
+	v4.Src().SetValue(ip4_1.Address())
+	v4.Dst().SetValue(ip4_2.Address())
+	v4.Priority().Dscp().Phb().SetValue(56)
+	otg.PushConfig(t, config)
+	otg.StartProtocols(t)
 
 	t.Log("Running traffic on DUT interfaces: ", dp1, dp2)
 	dutInPktsBeforeTraffic := dut.Telemetry().Interface(dp1.Name()).Counters().InUnicastPkts().Get(t)
 	dutOutPktsBeforeTraffic := dut.Telemetry().Interface(dp2.Name()).Counters().OutUnicastPkts().Get(t)
 	t.Log("inPkts and outPkts counters before traffic: ", dutInPktsBeforeTraffic, dutOutPktsBeforeTraffic)
-
-	ate.Traffic().Start(t, flow)
+	otg.StartTraffic(t)
 	time.Sleep(10 * time.Second)
-	ate.Traffic().Stop(t)
+	otg.StopTraffic(t)
 
 	ds1 := dut.Telemetry().Interface(dp1.Name()).OperStatus().Get(t)
 	if want := telemetry.Interface_OperStatus_UP; ds1 != want {
 		t.Errorf("Get(DUT port1 status): got %v, want %v", ds1, want)
 	}
-	as1 := ate.Telemetry().Interface(ap1.Name()).OperStatus().Get(t)
-	if want := telemetry.Interface_OperStatus_UP; as1 != want {
-		t.Errorf("Get(ATE port1 status): got %v, want %v", as1, want)
-	}
 	ds2 := dut.Telemetry().Interface(dp2.Name()).OperStatus().Get(t)
 	if want := telemetry.Interface_OperStatus_UP; ds2 != want {
 		t.Errorf("Get(DUT port2 status): got %v, want %v", ds2, want)
 	}
-	as2 := ate.Telemetry().Interface(ap2.Name()).OperStatus().Get(t)
-	if want := telemetry.Interface_OperStatus_UP; as2 != want {
-		t.Errorf("Get(ATE port2 status): got %v, want %v", as2, want)
+	// Verifying the ate port link state
+	for _, p := range config.Ports().Items() {
+		portMetrics := otg.Telemetry().Port(p.Name()).Get(t)
+		if portMetrics.GetLink() != otgtelemetry.Port_Link_UP {
+			t.Errorf("Get(ATE %v status): got %v, want %v", p.Name(), portMetrics.GetLink(), otgtelemetry.Port_Link_UP)
+		}
 	}
-	ateInPkts := ate.Telemetry().Flow(flow.Name()).Counters().InPkts().Get(t)
-	ateOutPkts := ate.Telemetry().Flow(flow.Name()).Counters().OutPkts().Get(t)
+
+	otgutils.LogFlowMetrics(t, otg, config)
+	ateInPkts := otg.Telemetry().Flow(flowName).Counters().InPkts().Get(t)
+	ateOutPkts := otg.Telemetry().Flow(flowName).Counters().OutPkts().Get(t)
 
 	if ateOutPkts == 0 {
-		t.Errorf("Get(out packets for flow %q: got %v, want nonzero", flow.Name(), ateOutPkts)
+		t.Errorf("Get(out packets for flow %q: got %v, want nonzero", flowName, ateOutPkts)
 	}
-	lossPct := ate.Telemetry().Flow(flow.Name()).LossPct().Get(t)
+	lossPct := float32((ateOutPkts - ateInPkts) * 100 / ateOutPkts)
```

Comment:
float32 conversion needs to happen before the division, otherwise it will already be too late. Alternatively, maybe we can keep it as an integer and below compare it as `if lossPct >= 1` just like elsewhere.

Reply:
Kept it as integer.